### PR TITLE
Adds odo dev command reference

### PR DESCRIPTION
**Pull request by cdrage**
**What type of PR is this:**

/kind documentation

**What does this PR do / why we need it:**

This PR adds documentation command reference for `odo dev` as well as
a new format / template for command referencing.

We go in detail with regards to the Devfile and explain everything
that's happening.

**Which issue(s) this PR fixes:**

Fixes https://github.com/redhat-developer/odo/issues/5584

**PR acceptance criteria:**

- [ ] Unit test

- [ ] Integration test

- [ ] Documentation

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

File: docs/website/versioned_docs/version-3.0.0/command-reference/dev.md
Comment (rm3l) [inline docs/website/versioned_docs/version-3.0.0/command-reference/dev.md:10]: Looks like this should not link to `localhost:3000`.

File: docs/website/versioned_docs/version-3.0.0/command-reference/dev.md
Comment (rm3l) [inline docs/website/versioned_docs/version-3.0.0/command-reference/dev.md:49]: I'm not a native English speaker, but it should be "is run", no? ^^

```suggestion
When `odo dev` is run, it first looks in the `devfile.yaml` for the instructions to be executed, specifically: `components` and `commands`.
```

File: docs/website/versioned_docs/version-3.0.0/command-reference/dev.md
Comment (rm3l) [inline docs/website/versioned_docs/version-3.0.0/command-reference/dev.md:100]: Same here.

```suggestion
> Build is what is initially run when deploying to the cluster.
```

**Review by rm3l — CHANGES_REQUESTED** [review]

File: docs/website/versioned_docs/version-3.0.0/command-reference/dev.md
Comment (cdrage) [reply docs/website/versioned_docs/version-3.0.0/command-reference/dev.md:10]: Woops! Good catch.

File: docs/website/versioned_docs/version-3.0.0/command-reference/dev.md
Comment (cdrage) [reply docs/website/versioned_docs/version-3.0.0/command-reference/dev.md:100]: Suppose to be ran :) since the sentence is past tense!

File: docs/website/versioned_docs/version-3.0.0/command-reference/dev.md
Comment (cdrage) [reply docs/website/versioned_docs/version-3.0.0/command-reference/dev.md:49]: Nevermind! It's suppose to be ran according to google.

**Review by rm3l — APPROVED** [review]

File: docs/website/versioned_docs/version-3.0.0/command-reference/dev.md
Comment (feloy) [inline docs/website/versioned_docs/version-3.0.0/command-reference/dev.md:59]:
```suggestion
A much more descriptive explanation on each part of a Devfile can be found on the [Devfile API reference](https://devfile.io/docs/devfile/2.0.0/user-guide/api-reference/) site.
```

File: docs/website/versioned_docs/version-3.0.0/command-reference/dev.md
Comment (feloy) [inline docs/website/versioned_docs/version-3.0.0/command-reference/dev.md:83]: Are you sure? I think this is the list of starter projects available "compatible" with the devfile

File: docs/website/versioned_docs/version-3.0.0/command-reference/dev.md
Comment (feloy) [inline docs/website/versioned_docs/version-3.0.0/command-reference/dev.md:96]: Not in the command line, but as a process in the container

File: docs/website/versioned_docs/version-3.0.0/command-reference/dev.md
Comment (feloy) [inline docs/website/versioned_docs/version-3.0.0/command-reference/dev.md:102]: This is executed after Build.

File: docs/website/versioned_docs/version-3.0.0/command-reference/dev.md
Comment (feloy) [inline docs/website/versioned_docs/version-3.0.0/command-reference/dev.md:104]: Seems straightforward, but it could be interesting to indicate that these commands are the commands needed to - build the program, - execute it in non-debug mode, - in debug mode, and - execute the tests.

**Review by feloy — CHANGES_REQUESTED** [review]

**Comment by cdrage** [comment]
Thanks for the review! I've gone ahead and updated the pr @feloy

**Review by feloy — APPROVED** [review]

**Comment by cdrage** [comment]
/approve

(only need 1 approval for docs, and @feloy is blocked on this PR as we need to push an `odo dev` reference doc out)

**Pull request merged**
